chia/plotters/bladebit.py

```diff
@@ -0,0 +1,166 @@
+            callback(line)
+        else:
+            break
+
+
+def parse_stdout(out):
+    out = out.rstrip()
+    print(out)
+    for k, v in progress.items():
+        if k in out:
+            print(f"Progress update: {v}")
+
+
+async def run(command):
+    process = await asyncio.create_subprocess_shell(
+        command, stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE
+    )
+
+    await asyncio.wait(
+        [
+            _read_stream(
+                process.stdout,
+                lambda x: parse_stdout(x.decode("UTF8")),
+            ),
+            _read_stream(
+                process.stderr,
+                lambda x: print("STDERR: {}".format(x.decode("UTF8"))),
+            ),
+        ]
+    )
+
+    await process.wait()
+
+
+async def run_bladebit(args):
+    cmd = ""
+    for arg in args:
+        cmd += arg
+        cmd += " "
+    print(f"Running command: {cmd[:-1]}")
+    await run(cmd[:-1])
+
+
+def install_bladebit(root_path):
+    if sys.platform.startswith("linux"):
+        print("Installing dependencies.")
+        try:
+            subprocess.run(
+                [
+                    "sudo",
+                    "apt",
+                    "install",
+                    "-y",
+                    "build-essential",
+                    "cmake",
+                    "libnuma-dev",
+                    "git",
+                ]
+            )
+        except Exception:
+            raise ValueError("Could not install dependencies.")
+
+        print("Cloning repository and its submodules.")
+        try:
+            subprocess.run(
+                [
+                    "git",
+                    "clone",
+                    "--recursive",
+                    "https://github.com/harold-b/bladebit.git",
+                ],
+                cwd=os.fspath(root_path),
+            )
+        except Exception:
+            raise ValueError("Could not clone bladebit repository.")
+
+        bladebit_path = os.fspath(root_path.joinpath("bladebit"))
+        print("Building BLS library.")
+        # Build bls library. Only needs to be done once.
+        try:
+            subprocess.run(["./build-bls"], cwd=bladebit_path)
+        except Exception as e:
+            raise ValueError(f"Building BLS library failed. {e}")
+
+        print("Build bladebit.")
+        try:
+            subprocess.run(["make", "clean"], cwd=bladebit_path)
+            subprocess.run(["make"], cwd=bladebit_path)
+        except Exception as e:
+            raise ValueError(f"Building bladebit failed. {e}")
+    else:
+        raise ValueError("Platform not supported yet for bladebit plotter.")
+
+
+def plot_bladebit(args, root_path):
+    if not os.path.exists(root_path / "bladebit/.bin/release/bladebit"):
+        print("Installing bladebit plotter.")
+        try:
+            install_bladebit(root_path)
+        except Exception as e:
+            print(f"Exception while installing madmax plotter: {e}")
```

Comment:
Copy & paste error:
`print(f"Exception while installing bladebit plotter: {e}")`

Reply:
Just to get it in the easily applied suggestion form.

```suggestion
            print(f"Exception while installing bladebit plotter: {e}")
```